File: EEProjects/GhoBlog/backend/ghoblog/src/main/java/org/oiakushev/ghoblog/service/CommentService.java
package org.oiakushev.ghoblog.service;

import org.oiakushev.ghoblog.dto.CommentRequest;
import org.oiakushev.ghoblog.model.Comment;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface CommentService {
    Page<Comment> getAllByArticleId(Long articleId, Pageable pageable);

    Optional<Comment> getById(Long id);

    Comment add(CommentRequest value);
}

File: EEProjects/GhoBlog/frontend/gho-blog/src/app/core/services/comment.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { PageData } from '../models/page-data.model';
import { ApiService } from './api.service';

@Injectable()
export class CommentService {
    constructor (
        private apiService: ApiService
      ) {}
    
    getAll(page = 0, size = 10): Observable<PageData> {
        return this.apiService.get(`/comments?page=${page}&size=${size}`);
    }

    getAllbyArticleId(page = 0, size = 10, articleId: number): Observable<PageData> {
        return this.apiService.get(`/comments?articleId=${articleId}&page=${page}&size=${size}`);
    }

    getItem(itemId: number): Observable<Comment> {
        return this.apiService.get(`/comments/${itemId}`);
    }

    add(item: any): Observable<Comment> {
        return this.apiService.post(`/comments`, item);
    }
}